clang-tools-extra/clang-tidy/bugprone/SizeofExpressionCheck.cpp

```diff
@@ -130,6 +140,14 @@ void SizeofExpressionCheck::registerMatchers(MatchFinder *Finder) {
                        this);
   }
 
+  if (WarnOnSizeOfInLoopTermination) {
+    Finder->addMatcher(LoopExpr(hasDescendant(binaryOperator(
```

Comment:
Since `hasDescendant` is used,  it also matches loop bodies, right? 
Maybe add a test:
```
int A[10];
for(int i = 0; i < sizeof(A); i++) {
         if (x < sizeof(A)) { ... } // Should this be matched? Will this be matched twice?
  }
```
Also please add some tests for nested loops.